src/mpid/ch4/src/mpidig_win.c

```diff
@@ -527,8 +524,21 @@ static int win_shm_alloc_impl(MPI_Aint size, int disp_unit, MPIR_Comm * comm_ptr
     } else
         total_shm_size = size;
 
-    /* try global symm heap only when multiple processes exist */
-    if (comm_ptr->local_size > 1) {
+    bool win_shared_ok = false;
+    if (comm_ptr->local_size == 1) {
+        win_shared_ok = true;
+    } else if (shm_comm_ptr && comm_ptr->local_size == shm_comm_ptr->local_size) {
+        win_shared_ok = true;
+    }
```

Comment:
I'm curious if `comm_ptr->node_count` be able to help make this determination simpler? Looking at these conditions, it's not immediately clear to me how they translate to whether or not to use shared memory.